marklogic-data-hub-central/ui/src/components/entities/create-edit-step-dialog/create-edit-step-dialog.test.tsx

```diff
@@ -63,9 +64,39 @@ describe('Create Edit Step Dialog component', () => {
 
         fireEvent.change(nameInput, { target: {value: ''}});
         expect(getByText('Name is required')).toBeInTheDocument();
-        expect(saveButton).toBeDisabled();
+        expect(saveButton).toBeEnabled();
+    });
+
+    test('Verify clicking "Save" with no collection shows error', async () => {
+      const { getByText, getByLabelText, getByPlaceholderText } = render(<CreateEditStepDialog {...data.newMerging} />);
+
+      fireEvent.change(getByPlaceholderText('Enter name'), { target: {value: ''}});
+  
+      fireEvent.click(getByText('Save'));
+  
+      // both messages should show when both boxes are empty
+      expect(getByText('Name is required')).toBeInTheDocument();
+      expect(getByText('Collection or Query is required')).toBeInTheDocument();
     });
 
+    test('Verify clicking "Save" with no name shows error', async () => {
```

Comment:
This test can be clubbed together with the previous one for collection as the expect statements are more or less the same and would avoid a re-render of the component to improve test performance.
Check my previous comment.

Reply:
fixed